back/dora/api/serializers.py

```diff
@@ -296,6 +300,17 @@ def get_frais_autres(self, obj):
     def get_profils(self, obj):
         return [c.name for c in obj.publics.all()]
 
+    def get_publics(self, obj):
+        all_items = [
+            item
+            for public in obj.publics.all()
+            for item in public.corresponding_di_publics
+        ]
+        return list(dict.fromkeys(all_items))
```

Comment:
Plutôt `set()` que `dict.fromkeys()` :
```suggestion
        all_items = [
            item
            for public in obj.publics.all()
            for item in public.corresponding_di_publics
        ]
        return list(set(all_items))
```
Voir directement une _set comprehension_ :
```suggestion
        all_items = {
            item
            for public in obj.publics.all()
            for item in public.corresponding_di_publics
        }
        return list(all_items)
```